jung-visualization/src/main/java/edu/uci/ics/jung/visualization/transform/Lens.java

```diff
@@ -40,13 +39,9 @@ public Lens(Dimension d) {
 
   /** @param d the size used for the lens */
   public void setSize(Dimension d) {
-
-    if (d.width <= 0 || d.height <= 0) {
-      d = VisualizationServer.DEFAULT_SIZE;
-    }
-    float ewidth = d.width / 1.5f;
-    float eheight = d.height / 1.5f;
-    lensShape.setFrame(d.width / 2 - ewidth / 2, d.height / 2 - eheight / 2, ewidth, eheight);
+    float width = d.width / 1.5f;
```

Comment:
Please add checks for verifying that the Dimension is valid:

    Preconditions.checkNotNull(d);
    Preconditions.checkArgument(d.width > 0, "width must be > 0");
    Preconditions.checkArgument(d.height > 0, "height must be > 0");